content/en/blog/_posts/2023-01-15-Security-Bahavior-Analysis/index.md

```diff
@@ -0,0 +1,87 @@
+
+Next, let's discuss why microservice architecture is well suited to security-behavior monitoring.
+
+## Security-Behavior of Microservices vs Monoliths
+
+Kubernetes is often used to support workloads designed with microservice architecture. By design, microservices aim to follow the UNIX philosophy of "Do One Thing And Do It Well". Each microservice has a bounded context and a clear interface. In other words, we can expect the microservice clients to send relatively regular requests and the microservice to present a relatively regular behavior as a response to these requests. Consequently, microservice architecture is an excellent candidate for security-behavior monitoring.
+
+{{< figure src="Microservices.png" alt="Image showing why microservices are well suited for security-behavior monitoring" class="diagram-large" caption="Figure 2. Microservices are well suited for security-behavior monitoring" >}}
+
+The diagram above clarifies how dividing a monolithic service to a set of microservices improves our ability to perform security-behavior monitoring and control. In a monolithic service approach, different client requests are intertwined, resulting in diminished ability to identify irregular client behaviors. Without prior knowledge, an observer of the intertwined client requests will find it hard to distinguish between types of requests and their related characteristics. Further, internal client requests are not exposed to the observer. Lastly, the aggregated behavior of the monolithic service is a compound of the many different internal behaviors of its components, making it hard to identify irregular service behavior.
+
+In a microservice environment, each microservice is expected by design to offer a more well-defined service and serve better defined type of requests. This makes it easier for an observer to identify irregular client behavior and irregular service behavior. Further, a microservice design exposes the internal requests and internal services which offer more security-behavior data to identify irregularities by an observer. Overall, this makes the microservice design pattern better suited for security-behavior monitoring and control.
+
+## Security-Behavior Under Kubernetes
+
+Kubernetes deployments seeking to add Security-Behavior may use [Guard](http://knative.dev/security-guard), developed under the CNCF project Knative. **Guard can be installed as a standalone tool to protect any HTTP-based Kubernetes service**. Alternatively, Guard is integrated into the full Knative automation suite that runs on top of Kubernetes.
+
+See:
+
+- [Guard on Github](https://github.com/knative-sandbox/security-guard) for using Guard as a standalone.
+- The Knative automation suite - Read about Knative, as an [Opinionated Kubernetes](https://davidhadas.wordpress.com/2022/08/29/knative-an-opinionated-kubernetes) implementation that simplifies and unifies the way web services are deployed on Kubernetes.
```

Comment:
How do we feel about mentioning Opinionated Kubernetes? Context matters. In David's own blog, that's more obviously David's own coining of a term. Mentioned on the Kubernetes blog, readers might infer that Opinionated Kubernetes is a concept of the project overall.

Reply:
Knative after all is a layer above Kubernetes that allows deploying services in a certain way... 
And it is opinionated as to how such services should be structured, deployed, managed, and even secured...

If this is a concern, we can change the sentence above to say: 

Read about Knative, in the blog post "[Opinionated Kubernetes](https://davidhadas.wordpress.com/2022/08/29/knative-an-opinionated-kubernetes)" which describes how Knative simplifies and unifies the way web services are deployed on Kubernetes.